demos/mqtt/mqtt_demo_lightweight/mqtt_demo_lightweight.c

```diff
@@ -85,139 +66,138 @@
  * The topic name starts with the client identifier to ensure that each demo
  * interacts with a unique topic name.
  */
-#define MQTT_EXAMPLE_TOPIC                   CLIENT_IDENTIFIER "/example/topic"
+#define MQTT_EXAMPLE_TOPIC           CLIENT_IDENTIFIER "/example/topic"
 
 /**
- * @brief Dimensions a file scope buffer currently used to send and receive MQTT data
- * from a socket.
+ * @brief Length of client MQTT topic.
  */
-#define SHARED_BUFFER_SIZE                   500U
+#define MQTT_EXAMPLE_TOPIC_LENGTH    ( ( uint16_t ) ( sizeof( MQTT_EXAMPLE_TOPIC ) - 1 ) )
+
+/**
+ * @brief Size of the network buffer for MQTT packets.
+ */
+#ifndef NETWORK_BUFFER_SIZE
+    #define NETWORK_BUFFER_SIZE    ( 1024U )
+#endif
 
 /**
  * @brief The MQTT message published in this example.
  */
-#define MQTT_EXAMPLE_MESSAGE                 "Hello Light Weight MQTT World!"
+#define MQTT_EXAMPLE_MESSAGE                 "Hello World!"
 
 /**
  * @brief Keep alive period in seconds for MQTT connection.
  */
-#define MQTT_KEEP_ALIVE_PERIOD_SECONDS       5U
+#define MQTT_KEEP_ALIVE_INTERVAL_SECONDS     ( 5U )
 
 /**
  * @brief Socket layer transportTimeout in milliseconds.
  */
-#define TRANSPORT_SEND_RECV_TIMEOUT_MS       200U
+#define TRANSPORT_SEND_RECV_TIMEOUT_MS       ( 200U )
 
 /**
  * @brief Number of time network receive will be attempted
  * if it fails due to transportTimeout.
  */
-#define MQTT_MAX_RECV_ATTEMPTS               10U
+#define MQTT_MAX_RECV_ATTEMPTS               ( 10U )
 
 /**
  * @brief Delay between two demo iterations.
  */
-#define MQTT_DEMO_ITERATION_DELAY_SECONDS    5U
+#define MQTT_DEMO_ITERATION_DELAY_SECONDS    ( 5U )
 
 /*-----------------------------------------------------------*/
 
 /**
- * @brief Establish a TCP connection to the given server.
+ * @brief connect to MQTT broker with reconnection retries.
+ * If connection fails, retry is attempted after a timeout.
+ * Timeout value will exponentially increase until maximum
+ * timeout value is reached or the number of attempts are exhausted.
```

Comment:
(Nit) Should this say "..until maximum reconnection delay time is reached..." 
Use of _timeout_ can be confusing as if relating tcp connect timeout.
Possibly, the macro naming and documentation in the reconnect file should be also updated (which can be done in a separate PR)